sdk/trace-shaded-deps/src/main/java/io/opentelemetry/sdk/trace/internal/JcTools.java

```diff
@@ -23,38 +18,19 @@
  */
 public final class JcTools {
 
-  private static final AtomicBoolean queueCreationWarningLogged = new AtomicBoolean();
-  private static final Logger logger = Logger.getLogger(JcTools.class.getName());
-
   /**
    * Returns a new {@link Queue} appropriate for use with multiple producers and a single consumer.
    */
   public static <T> Queue<T> newFixedSizeQueue(int capacity) {
-    try {
-      return new MpscArrayQueue<>(capacity);
-    } catch (java.lang.NoClassDefFoundError | java.lang.ExceptionInInitializerError e) {
-      if (!queueCreationWarningLogged.getAndSet(true)) {
-        logger.log(
-            Level.WARNING,
-            "Cannot create high-performance queue, reverting to ArrayBlockingQueue ({0})",
-            Objects.toString(e, "unknown cause"));
-      }
-      // Happens when modules such as jdk.unsupported are disabled in a custom JRE distribution,
-      // or a security manager preventing access to Unsafe is installed.
-      return new ArrayBlockingQueue<>(capacity);
-    }
+    return new MpscAtomicArrayQueue<>(capacity);
```

Comment:
I originally only used this implementation on Java 22+, to avoid triggering the Unsafe warning

but given that the benchmarks look fine, I think it would be ok to go straight to this implementation in all cases and simplify things

Reply:
they're pretty similar, in fact looks like MpscAtomicArrayQueue is autogenerated from MpscArrayQueue:

https://github.com/JCTools/JCTools/blob/a17b56f09cc9ec7d773707578c89f3ca9c11f968/jctools-core/src/main/java/org/jctools/queues/atomic/MpscAtomicArrayQueue.java#L20-L23